### PR TITLE
ISSUE-2223: Fixed the bug with breadcrumbs on PDP with a product without categories

### LINKED ISSUE #2223: [bug]: The breadcrumbs have the bug if the product assigned only "default" category or isn't assigned anything category.
**Describe the bug**
There is a problem in the talon "useProductFullDetail".
If a product isn't assigned anything category or assigned only "default" category, the system is showing the error and doesn't render frontend when a client tries to view the product by link.
Please take a look at the screenshots.

**To reproduce**
1. Unassign the categories from the product using the admin panel;
2. Copy the product link using URL-Key;
3. Try to view the product by link;
4. See error below;
5. Also, you can check this error in the console

**Expected behavior**
PDP must be rendered without error.

**Screenshots**
The error on the storefront 
error-on-front

The error in the console
error-on-console

**Additional context**

**Possible solutions**
I have already fixed it on my project and I am going to create a PR here.

**Please complete the following device information:**
 - Device [e.g. iPhone6, PC, Mac, Pixel3]: All
 - OS [e.g. iOS8.1, Windows 10]: All
 - Browser [e.g. Chrome, Safari]: All
 - Browser Version [e.g. 22]:
 - Magento Version: 2.3.4
 - PWA Studio Version:
 - NPM version `npm -v`:
 - Node Version `node -v`:

**Please let us know what packages this bug is in regards to:**
- [ ] `venia-concept`
- [ ] `venia-ui`
- [ ] `pwa-buildpack`
- [ ] `peregrine`
- [ ] `pwa-devdocs`
- [ ] `upward-js`
- [ ] `upward-spec`
- [ ] `create-pwa`

### PR DESCRIPTION
**Pull request by andrewbess**
## Description

This PR fixes the bug with breadcrumbs on PDP when a product isn't assigned anything category

## Related Issue

Closes #2223.

### Verification Steps

1. Go to the admin panel in backend instance;
2. Go to the product grid and select any product to edit;
3. Unassign the categories from the product and save a product;
4. Run reindex;
5. Copy the product link using URL-Key;
6. Try to view the product by link;
7. Verify the error doesn't reproduce.

## Screenshots / Screen Captures (if appropriate)
error-on-console

### CODE REVIEW

File: packages/peregrine/lib/talons/ProductFullDetail/useProductFullDetail.js [hunk 1]
```diff
@@ -97,6 +97,9 @@ const getMediaGalleryEntries = (product, optionCodes, optionSelections) => {
 // product has multiple related categories. This function filters and selects
 // one category id for that purpose.
 const getBreadcrumbCategoryId = categories => {
+    if (!!categories.length === false) {
```
Comment (sirugh) [inline packages/peregrine/lib/talons/ProductFullDetail/useProductFullDetail.js]: I'm guessing `categories` is an empty array when we get here and there is no category assigned to a product. In that case, let's just use a simple check against truthyness of `categories` and check whether there are any values in `categories` array, if it exists:

```js
// Exit if there are no categories for this product.
if (!categories || !categories.length) {
  return;
}
```

Comment (jimbo) [reply packages/peregrine/lib/talons/ProductFullDetail/useProductFullDetail.js]: @andrewbess Thanks for contributing! Looks like a simple fix. 👍 

@sirugh Agreed, good suggestion.

Comment (andrewbess) [reply packages/peregrine/lib/talons/ProductFullDetail/useProductFullDetail.js]: Hello @sirugh 
Thank you for review.
I have fixed it.
Please check.

**Comment by dpatil-magento** [comment]
Looks good.

image

**Pull request merged**